### PR TITLE
In Android O, some IntentServices will not trigger if the app has bee…

### PR DESCRIPTION
**Pull request by frankgh**
…n stopped. To fix this we do the following:

- Add a BroadcastReceiver that will receive the intent and call a JobIntentService to handle the intent
- Use JobIntentService (Added for Android O), instead of IntentService
- Set the geofence to never expire
- For pendingIntent, we getBroadcast instead of getService

### CODE REVIEW

File: Geofencing/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -15,29 +14,37 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.google.android.gms.location.sample.geofencing" >
+    package="com.google.android.gms.location.sample.geofencing">
 
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
Comment (Trollologic) [inline Geofencing/app/src/main/AndroidManifest.xml]: When running on Android O, the JobScheduler will take care of wake locks but on previous versions of the platform you must request the WAKE_LOCK permission.

**Comment by frankgh** [comment]
Adding WAKE_LOCK permission

File: Geofencing/app/src/main/java/com/google/android/gms/location/sample/geofencing/Constants.java [hunk 2]
```diff
@@ -42,8 +43,8 @@ private Constants() {
     /**
      * For this sample, geofences expire after twelve hours.
      */
-    static final long GEOFENCE_EXPIRATION_IN_MILLISECONDS =
-            GEOFENCE_EXPIRATION_IN_HOURS * 60 * 60 * 1000;
+    static final long GEOFENCE_EXPIRATION_IN_MILLISECONDS = Geofence.NEVER_EXPIRE;
+            //GEOFENCE_EXPIRATION_IN_HOURS * 60 * 60 * 1000;
```
Comment (samtstern) [inline Geofencing/app/src/main/java/com/google/android/gms/location/sample/geofencing/Constants.java]: Can you explain why you have moved this to never expire?  If this change is intentional we need to remove the commented code and update the JavaDoc comment above.

Comment (frankgh) [reply Geofencing/app/src/main/java/com/google/android/gms/location/sample/geofencing/Constants.java]: The idea was to showcase that the Geofences will trigger even if the app has been killed. If the OS decides to kill the app, the notification should still trigger after entering/leaving the Geofence. One can manually kill the app and verify that the behavior is correct.

I suggest reverting to the original 12 hour expiration. I will go ahead and push that change.

Thank you,

Comment (frankgh) [reply Geofencing/app/src/main/java/com/google/android/gms/location/sample/geofencing/Constants.java]: @samtstern done, the 12 hour expiration is restored

**Comment by samtstern** [comment]
@frankgh thanks for contributing!

**Pull request merged**